src/System.Private.CoreLib/shared/System/Buffers/Text/Utf8Parser/ParserHelpers.cs

```diff
@@ -65,7 +66,7 @@ public static bool TryParseThrowFormatException(out int bytesConsumed)
         //
         // Enable use of ThrowHelper from TryParse() routines without introducing dozens of non-code-coveraged "value= default; bytesConsumed = 0; return false" boilerplate.
         //
-        public static bool TryParseThrowFormatException<T>(out T value, out int bytesConsumed)
+        public static bool TryParseThrowFormatException<T>(out T value, out int bytesConsumed) where T : struct
```

Comment:
Here and one other place I added the struct constraint, which will impact the generated IL, but not negatively, and it seemed like the correct fix.